src/Symfony/Component/ClassLoader/ClassCollectionLoader.php

```diff
@@ -106,6 +106,7 @@ public static function load($classes, $cacheDir, $name, $autoReload, $adaptive =
 
         $c = '(?:\s*+(?:(?:#|//)[^\n]*+\n|/\*(?:(?<!\*/).)++)?+)*+';
         $strictTypesRegex = str_replace('.', $c, "'^<\?php\s.declare.\(.strict_types.=.1.\).;'is");
+        $haltCompilerRegex = str_replace('.', $c, "'\b__halt_compiler.\(.\)'is");
```

Comment:
I've read the other issues and I think you are correct, also `__DIR__` and `__FILE__` should not be inlined.
Since the `ClassCollectionLoader` main purpose (AFAIK) is to avoid calling the autoloader for common classes, I don't think that we have major performance issue with a bunch of added `fstat`/`include`s (which can be `opcache`d btw).

I think the regex you suggest deserve to be multine to be a little more readable, but I'll see how much effort it requires.